docs/gallery/advanced_io/plot_editing.py

```diff
@@ -0,0 +1,178 @@
+"""
+Editing NWB files
+=================
+
+This tutorial demonstrates how to edit NWB files. How and whether it is possible to edit
+an NWB file depends on the storage backend and the type of edit. Here, we go through the
+common types of edits for HDF5 files. Keep in mind that any edit to an existing NWB file
+make it no longer a valid NWB file. We call this "doing surgery" on the NWB file.
+We highly recommend making a copy before editing and running a validation check on the
+file after editing it. See :ref:`validating`.
+
+In-place editing with h5py
+---------------------------
+The :py:mod:`h5py` package allows for in-place editing of HDF5 files. Where possible,
+this is the recommended way to edit HDF5 files.
+
+Editing values of datasets and attributes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can  change the value(s) of a dataset using :py:mod:`h5py`.
+
+First, let's create an NWB file with data:
+"""
+from pynwb import NWBHDF5IO, NWBFile, TimeSeries
+from datetime import datetime
+from dateutil.tz import tzlocal
+import numpy as np
+
+nwbfile = NWBFile(
+    session_description="my first synthetic recording",
+    identifier="EXAMPLE_ID",
+    session_start_time=datetime.now(tzlocal()),
+    session_id="LONELYMTN",
+)
+
+nwbfile.add_acquisition(
+    TimeSeries(
+        name="synthetic_timeseries",
+        description="Random values",
+        data=np.random.randn(100, 100),
+        unit="m",
+        rate=10e3,
+    )
+)
+
+with NWBHDF5IO("test_edit.nwb", "w") as io:
+    io.write(nwbfile)
+
+
+##############################################
+# Now, let's try to edit the values of the dataset:
+import h5py
+
+with h5py.File("test_edit.nwb", "r+") as f:
+    f["acquisition"]["synthetic_timeseries"]["data"][:10] = 0.0
```

Comment:
For datasets we can do this PyNWB directly. @rly does changing values also work for attributes? If datasets and attributes can be edited directly from PyNWB, then I think we should make that the recommended way,  rather than h5py.

Reply:
I guess technically you could do e.g.

```python
nwbfile.acquisition["time_series"].data.attrs["unit"] = "volts"
```
but this will not work:
```python
nwbfile.acquisition["time_series"].data.units = "volts"
```

nor 

```python
nwbfile.acquisition["time_series"].units = "volts"
```